tests/components/influxdb/test_init_v2.py

```diff
@@ -0,0 +1,749 @@
+"""The tests for the InfluxDB component."""
+import datetime
+import unittest
+from unittest import mock
+
+import homeassistant.components.influxdb as influxdb
+from homeassistant.const import (
+    EVENT_STATE_CHANGED,
+    STATE_OFF,
+    STATE_ON,
+    STATE_STANDBY,
+    UNIT_PERCENTAGE,
+)
+from homeassistant.setup import setup_component
+
+from tests.common import get_test_home_assistant
+
+
+@mock.patch("homeassistant.components.influxdb.InfluxDBClientV2")
+@mock.patch(
+    "homeassistant.components.influxdb.InfluxThread.batch_timeout",
+    mock.Mock(return_value=0),
+)
+class TestInfluxDB(unittest.TestCase):
```

Comment:
New tests must be standalone pytest test functions.

Reply:
Not sure I understand this one. Is this because I used the same class name in `test_init_v2.py`?  Would it also be resolved when I merged `test_init.py` with `test_init_v2.py` as you suggested?

Reply:
No, we just have that as a standard.

Reply:
@MartinHjelmare can you clarify what you mean here or point me to something that describes the standard you're referring to? Sorry I'm still learning python and I'm not seeing anything in the developer docs about this so trying to figure it out.

Or is there a test suite I can look at as a good example to follow?  I tried to follow the pattern I was seeing in the existing test file for influx but it sounds like that's not necessarily using the current best practices.

Reply:
Here's an example:
https://github.com/home-assistant/core/blob/ad5101c5c035ff5a8674e74fb98c7cc6d6927432/tests/components/manual/test_alarm_control_panel.py#L88-L125

Reply:
I see, so break it up into individual independent methods instead of how it is right now where there's a class with a common setup and a mock passed in to each test. This will likely make it easier to parameterize the tests like you suggested in your other feedback though so it seems like a good idea. Hopefully won't take too long once I get one or two down and figure out the pattern, will get to work on it.

Reply:
@MartinHjelmare Ok so that took longer then expected but I believe I completed it. All tests have now been merged into a single file and are using individual, parameterized pytests. Please take a look when you get a chance.

One issue I have now though is it looks like @balloob changed the existing test file to stop using `tearDown` and start using `tear_down_cleanup`. This created a conflict for my change. Since I completely re-wrote the test file to using individual pytests instead of using `unittest.TestCase` how do I incorporate this change?  Is there a pytest equivalent I should use? Or is that something that's only relevant when using `unittest.TestCase` classes and can be ignored now that its individual pytest methods?

Reply:
Yeah, `tear_down_cleanup` is only relevant for `unittest.TestCase` tests. So we don't need to keep it now.

Reply:
Ok cool so I'll resolve that conflict then, sounds good.